sources/core/Stride.Core.CompilerServices/Analyzers/STRDIAG010InvalidConstructor.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Stride.Core.CompilerServices.Common;
+
+namespace Stride.Core.CompilerServices.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class STRDIAG010InvalidConstructor : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "STRDIAG010";
+    private const string Title = "Invalid Constructor";
+    private const string MessageFormat = "The Type '{0}' doesn't have an empty constructor, which is needed for Serialization";
```

Comment:
If I'm right about public below, should be updated here. And I think the phrase is `public parameterless constructor`.

Reply:
it has to be public , internal doesnt count, the editor wont find the type if its internal

Reply:
is updated